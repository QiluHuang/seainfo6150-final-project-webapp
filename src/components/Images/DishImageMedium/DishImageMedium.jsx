import React from "react";
import PropTypes from "prop-types";
import styles from "./DishImageMedium.module.css";

const DishImage = (props) => {
    return (
        <div>
            {/* <div className={styles.image}> </div> */}
            <img className={styles.img} src={props.url} alt={props.title} />
        </div>
        
    )
}

DishImage.prototype = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
};
export default DishImage;