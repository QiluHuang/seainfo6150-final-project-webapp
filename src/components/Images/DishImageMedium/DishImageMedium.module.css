.img {
    width: 100%;
    height: 10vw;
    object-fit: cover;
    object-position: center;
    border-radius: 10px;
}

@media screen and (min-width: 525px) {
    .image {
      width: 65%; 
    }
  }
  
  @media screen and (min-width: 800px) {
    .image {
      width: 50%;
    }
  }
  
  @media screen and (min-width: 1150px) {
    .image {
        width: 45%;
    }
  }
  