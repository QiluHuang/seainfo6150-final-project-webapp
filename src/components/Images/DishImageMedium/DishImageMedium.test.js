import React from "react";
import { render } from "@testing-library/react";
import DishImageMedium from "./DishImageMedium.jsx";

const imageURL = "https://d.newsweek.com/en/full/1552244/variety-chinese-food.jpg";
const title = "soup";

describe("DishImageMedium tests", () => {
    it("renders the DishImageMedium component correctly", () => {
    const { container } = render(<DishImageMedium url={imageURL} title={title} />);
    expect(container).toMatchSnapshot();
    });
});