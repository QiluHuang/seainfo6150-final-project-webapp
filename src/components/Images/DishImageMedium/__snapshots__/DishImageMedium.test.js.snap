// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DishImageMedium tests renders the DishImageMedium component correctly 1`] = `
<div>
  <div>
    <img
      alt="soup"
      class="img"
      src="https://d.newsweek.com/en/full/1552244/variety-chinese-food.jpg"
    />
  </div>
</div>
`;
